### ISSUE TITLE
Documentation says ares_options.udp_port is in network byte order

### ISSUE THREAD

**Issue by aaron-michaux** [c-ares/c-ares#487]
The documentation says:

```
ARES_OPT_UDP_PORT unsigned short udp_port;
The port to use for queries over UDP, in network byte order. The default value is 53 (in network byte order), the standard name service port.
```

Yet, the code says:

```
  // NOTE: c-ares is assuming `options->udp_port` is in host byte order!
  if ((optmask & ARES_OPT_UDP_PORT) && channel->udp_port == -1)
    channel->udp_port = htons(options->udp_port);
```

**Comment by bradh352** [comment]
I haven't looked into this, but did you confirm via testing the docs are wrong, or did you just happen to glance at the code?

The reason I ask is there are quite a few usages of htons and ntohs in the c-ares code involving ports, so it could be swapping twice which would mean the docs could be right.

**Comment by aaron-michaux** [comment]
Hi,

I tested the code, I read the code, and I read the docs. I'm pretty sure that the docs are wrong.

**Comment by bradh352** [comment]
fixed in a306ed4
